tests/this_param/interface.js

```diff
@@ -0,0 +1,89 @@
+/*@flow*/interface I {
+  anotherMethod(): number;
+  static anotherStaticMethod(): number;
+}
+
+class A {
+  x: number;
+  static x: number;
+  method(this: this & I): number {
+    return this.x + this.anotherMethod();
+  }
+  static staticMethod(this: Class<I> & Class<this>): number {
```

Comment:
Under `this: Class<this> & Class<I>`, the `this`-fixing process leaves a tvar flowing to SpeculativeMatchT instead of a concrete type.  When the `Class<this>` finally goes concrete, the speculative testing setup has already unwound, and the error leaks.

Swapping the intersection members like they're configured here allows the method to be found on `Class<I>` without ever touching `Class<this>`, hence the absence of an error in the current configuration.

Quick and dirty hypothesis check--the following eliminates the spurious error:

```ocaml
    (*Above the (head_l, SpeculativeMatchT) case (calls `speculative_match head_l u`)*)
    | (ClassT instance, UseT (_, SpeculativeMatchT _)) -> (
      let reason = prefix_reason "statics of " (reason_of_t instance) in
      let tvar = mk_tvar cx reason in
      rec_flow cx trace (instance, GetPropT(reason, (reason, "statics"), tvar));
      rec_flow cx trace (tvar, ReposLowerT (reason, u))
    )
```

```ocaml
    (*Above the (ClassT, _) flows statics to u*)
    | (InstanceT _, UseT (k, ClassT instance)) ->
      let reason = prefix_reason "statics of " (reason_of_t instance) in
      let tvar = mk_tvar cx reason in
      rec_flow cx trace (instance, GetPropT(reason, (reason, "statics"), tvar));
      rec_flow cx trace (l, UseT(k,tvar))
```